extensions/wikia/AdEngine/js/lookup/a9.js

```diff
@@ -0,0 +1,164 @@
+/*global define*/
+define('ext.wikia.adEngine.lookup.a9', [
+	'ext.wikia.adEngine.context.slotsContext',
+	'ext.wikia.adEngine.lookup.lookupFactory',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (slotsContext, factory, doc, log, win) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.lookup.a9',
+		config = {
+			oasis: {
+				BOTTOM_LEADERBOARD: [[728, 90], [970, 250]],
+				INCONTENT_BOXAD_1: [[300, 250], [300, 600]],
+				TOP_LEADERBOARD: [[728, 90], [970, 250]],
+				TOP_RIGHT_BOXAD: [[300, 250], [300, 600]]
+			},
+			mercury: {
+				MOBILE_IN_CONTENT: [[300, 250], [320, 480]],
+				MOBILE_PREFOOTER: [[320, 50], [300, 250]],
+				MOBILE_TOP_LEADERBOARD: [[320, 50]]
+			}
+		},
+		amazonId = '3115',
+		bids = {},
+		loaded = false,
+		priceMap = {},
+		slots = [];
+
+	function call(skin, onResponse) {
+		var a9Script = doc.createElement('script'),
+			node = doc.getElementsByTagName('script')[0],
+			a9Slots;
+
+		if (!loaded) {
+			log(['call - load', skin], 'debug', logGroup);
+
+			a9Script.type = 'text/javascript';
+			a9Script.async = true;
+			a9Script.src = '//c.amazon-adsystem.com/aax2/apstag.js';
+
+			node.parentNode.insertBefore(a9Script, node);
+
+			configureApstag();
+
+			win.apstag.init({
+				pubID: amazonId
+			});
+
+			loaded = true;
+		}
+
+		slots = slotsContext.filterSlotMap(config[skin]);
+		a9Slots = getA9Slots(slots);
+		log(['call - fetchBids', a9Slots], 'debug', logGroup);
+
+		win.apstag.fetchBids({
+			slots: a9Slots,
+			timeout: 2000
+		}, function(currentBids) {
+			log(['call - fetchBids response', currentBids], 'debug', logGroup);
+
+			currentBids.forEach(function (bid) {
+				bids[bid.slotID] = bid;
```

Comment:
Done in https://github.com/Wikia/app/pull/13603/commits/2d523760307796875dadae429ac8bc34e9f5533b